src/frontend/web_application/src/scenes/Contact/components/PublicKeyForm/presenter.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { withI18n, Trans } from '@lingui/react';
+import renderReduxField from '../../../../services/renderReduxField';
+import { Button, Confirm, FieldErrors, Fieldset, FormColumn, FormGrid, FormRow, Icon, Legend, TextFieldGroup as TextFieldGroupBase, TextareaFieldGroup as TextareaFieldGroupBase } from '../../../../components';
+
+import './style.scss';
+
+const TextFieldGroup = renderReduxField(TextFieldGroupBase);
+const TextareaFieldGroup = renderReduxField(TextareaFieldGroupBase);
+
+@withI18n()
+class PublicKeyForm extends PureComponent {
+  static propTypes = {
+    errors: PropTypes.arrayOf(PropTypes.string),
+    i18n: PropTypes.shape({}).isRequired,
+    contactId: PropTypes.string.isRequired,
+    publicKey: PropTypes.shape({}),
+    handleSubmit: PropTypes.func.isRequired,
+    form: PropTypes.string.isRequired,
+    createPublicKey: PropTypes.func.isRequired,
+    updatePublicKey: PropTypes.func.isRequired,
+    deletePublicKey: PropTypes.func.isRequired,
+    onSuccess: PropTypes.func,
+    onCancel: PropTypes.func,
+  };
+
+  static defaultProps = {
+    publicKey: undefined,
+    onSuccess: undefined,
+    onCancel: undefined,
+    errors: [],
+  };
+
+  handleSubmit = (ev) => {
+    const {
+      contactId, handleSubmit, publicKey,
+      createPublicKey, updatePublicKey,
+      onSuccess,
+    } = this.props;
+
+    try {
+      if (publicKey) {
+        const { key_id: publicKeyId, label } = publicKey;
+
+        handleSubmit(ev)
+          .then(values => updatePublicKey({
+            contactId,
+            publicKey: { ...values, publicKeyId },
+            original: { label },
+          }));
+      } else {
+        handleSubmit(ev)
+          .then(values => createPublicKey({ contactId, publicKey: values }));
+      }
+    } catch (ex) {
+      return;
```

Comment:
may be notify something went wrong or display error.

If it is used to prevent an unpredictable error, I prefer to not fail silently but crash with our magnificent BSOD.